merino-web/src/providers.rs

```diff
@@ -98,6 +98,26 @@ async fn make_provider_tree(
             StealthProvider::new_boxed(inner)
         }
 
+        SuggestionProviderConfig::ClientVariantSwitch(filter_config) => {
+            let matching_provider = make_provider_tree(
+                settings,
+                filter_config.matching_provider.as_ref(),
+                metrics_client,
+            )
+            .await?;
+            let default_provider = make_provider_tree(
+                settings,
+                filter_config.default_provider.as_ref(),
+                metrics_client,
+            )
+            .await?;
+            ClientVariantFilterProvider::new_boxed(
+                matching_provider,
+                default_provider,
+                (*filter_config.client_variant).to_string(),
```

Comment:
I think this might be clearer written as
```suggestion
                filter_config.client_variant.clone(),
```